### PR TITLE
Add Windows support for File and Service modules

### PR DESCRIPTION
**Pull request by tdas3001**
Opened a new PR for adding Windows support for File and Service modules in Testinfra as the old one  #532 has been inactive for quite some time. The PR contains all the changes in #532 and a few more implementations.

```py
# Basic file and service operations are now supported in windows remote server
>>> host.file("C:/Users/Administrator/pytest-testinfra").is_directory
True
>>> host.file("C:/Users/Administrator/pytest-testinfra/images").listdir()
['alphine', 'archlinux', 'debian_bullseye', 'rockylinux8', 'ubuntu_xenial']

```
Please let me know of any comments.

### CODE REVIEW

**Comment by tdas3001** [comment]
@philpep feel free to have a look at the PR

File: testinfra/modules/file.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 # limitations under the License.
 
 import datetime
+import six
```
Comment (philpep) [inline testinfra/modules/file.py:14]:
```suggestion
```

File: testinfra/modules/file.py [hunk 2]
```diff
@@ -312,3 +315,200 @@ class NetBSDFile(BSDFile):
...
+            self.path,
+        )
+
+    @property
+    def user(self):
+        """Return file owner as string
+
+        >>> host.file("C:/Windows/passwd").user
+        'root'
+        """
```
Comment (philpep) [inline testinfra/modules/file.py:392]:
```suggestion
```

File: testinfra/modules/file.py [hunk 3]
```diff
@@ -312,3 +315,200 @@ class NetBSDFile(BSDFile):
...
+        raise NotImplementedError
+
+    @property
+    def uid(self):
+        """Return file user id as integer
+
+        >>> host.file("C:/Windows/passwd").uid
+        0
+        """
```
Comment (philpep) [inline testinfra/modules/file.py:401]:
```suggestion
```

File: testinfra/modules/file.py [hunk 4]
```diff
@@ -312,3 +315,200 @@ class NetBSDFile(BSDFile):
...
+        raise NotImplementedError
+
+    @property
+    def sha256sum(self):
+        raise NotImplementedError
+
+    def _get_content(self, decode):
+        out = self.run_test(r"powershell -command \"cat -- '%s'\"", self.path)
+        if out.rc != 0:
+            raise RuntimeError("Unexpected output %s" % (out,))
```
Comment (philpep) [inline testinfra/modules/file.py:441]:
```suggestion
        out = self.run_expect([0], r"powershell -command \"cat -- '%s'\"", self.path)
```

File: testinfra/modules/file.py [hunk 5]
```diff
@@ -312,3 +315,200 @@ class NetBSDFile(BSDFile):
...
+        >>> host.file("C:/Windows/Temp").listdir()
+        ['foo_file', 'bar_dir']
+        """
+        out = self.run_test(
+            r"powershell -command \"Get-ChildItem -Path '%s' | Select-Object -ExpandProperty Name\"",
+            self.path,
+        )
+        if out.rc != 0:
+            raise RuntimeError("Unexpected output {}".format(out))
+        return [item.strip() for item in out.stdout.strip().split("\n")]
```
Comment (philpep) [inline testinfra/modules/file.py:501]:
```suggestion
        out = self.check_output(
            r"powershell -command \"Get-ChildItem -Path '%s' | Select-Object -ExpandProperty Name\"",
            self.path,
        )
        return [item.strip() for item in out.strip().split("\n")]
```

File: testinfra/modules/file.py [hunk 6]
```diff
@@ -312,3 +315,200 @@ class NetBSDFile(BSDFile):
...
+
+    def __eq__(self, other):
+        if isinstance(other, File):
+            return self.path == other.path
+        if isinstance(other, six.string_types):
+            return self.path == other
+        return False
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
```
Comment (philpep) [inline testinfra/modules/file.py:514]:
```suggestion
```

(this is already defined in parent class)

File: testinfra/modules/service.py [hunk 7]
```diff
@@ -275,3 +277,25 @@ def is_running(self):
...
     def is_enabled(self):
         raise NotImplementedError
+
+
+class WindowsService(Service):
+    @property
+    def is_running(self):
+        return (
+            self.check_output(
+                "Get-Service '%s' | Select -ExpandProperty Status",  # noqa: E501
```
Comment (philpep) [inline testinfra/modules/service.py:287]:
```suggestion
                "Get-Service '%s' | Select -ExpandProperty Status",
```

useless noqa ?

File: testinfra/modules/service.py [hunk 8]
```diff
@@ -275,3 +277,25 @@ def is_running(self):
...
+                self.name,
+            )
+            == "Running"
+        )
+
+    @property
+    def is_enabled(self):
+        return (
+            self.check_output(
+                "Get-Service '%s' | Select -ExpandProperty StartType",  # noqa: E501
```
Comment (philpep) [inline testinfra/modules/service.py:297]:
```suggestion
                "Get-Service '%s' | Select -ExpandProperty StartType",
```

useless noqa ?

**Review by philpep — CHANGES_REQUESTED** [review]
Thanks! Like I said in some other PR, I don't have knowledge neither time to maintain this. However this PR looks almost reasonable, so if other windows user can test this and tell me it's useful for them, I'll be happy to merge it.

**Comment by tdas3001** [comment]
> Thanks! Like I said in some other PR, I don't have knowledge neither time to maintain this. However this PR looks almost reasonable, so if other windows user can test this and tell me it's useful for them, I'll be happy to merge it.

Thanks for your comments @philpep . I would get back to them asap!

**Review by philpep — APPROVED** [review]

**Comment by philpep** [comment]
Merged, thanks!

**Pull request merged**